include/redis/Redis.h

```diff
@@ -123,7 +123,7 @@ namespace redis {
         *  ------ Locking ------
         */
 
-        long acquireLock(const std::string &key, int expirySeconds);
```

Comment:
I've not got anything against `std::optional`, but it feels a bit out of sync with the rest of the codebase. Elsewhere I think we take the approach of "return the value if it's worked, and -1 or zero if not". If this is just to avoid the cast from `uint32_t` to `long`, could we just change this return value to `uint32_t` and say `0` means it's failed?